File: example/javalib/testing/1-testing/test/src/qux/QuxTests.java
package qux;

import static org.junit.jupiter.api.Assertions.assertTrue;

import org.junit.jupiter.api.Test;

public class QuxTests {

  public static void assertHello(String result) {
    assertTrue(result.startsWith("Hello"));
  }

  public static void assertWorld(String result) {
    assertTrue(result.endsWith("World"));
  }

  @Test
  public void hello() {
    String result = Qux.hello();
    assertHello(result);
  }

  @Test
  public void world() {
    String result = Qux.hello();
    assertWorld(result);
  }
}

File: example/scalalib/testing/1-testing/test/src/QuxTests.scala
package qux
import utest.*
object QuxTests extends TestSuite {
  def assertHello(result: String): Unit = {
    assert(result.startsWith("Hello"))
  }

  def assertWorld(result: String): Unit = {
    assert(result.endsWith("World"))
  }

  def tests = Tests {
    test("hello") {
      val result = Qux.hello()
      assertHello(result)
      result
    }
    test("world") {
      val result = Qux.hello()
      assertWorld(result)
      result
    }
  }
}
